spec/unit/persistent_enum_spec.rb

```diff
@@ -0,0 +1,379 @@
+require 'persistent_enum'
+require 'byebug'
+
+RSpec.describe PersistentEnum, :database do
+  CONSTANTS = [:One, :Two, :Three, :Four]
+
+  before(:context) do
+    initialize_database
+  end
+
+  after(:each) do
+    destroy_test_models
+  end
+
+  shared_examples "acts like an enum" do
+    # abstract :model
+
+    it "looks up each value" do
+      CONSTANTS.each do |c|
+        e = model.value_of(c)
+        expect(e).to               be_present
+        expect(e.enum_constant).to be_a(String)
+        expect(e.to_sym).to        eq(c)
+        expect(e).to               eq(model[e.ordinal])
+        expect(e).to               eq(model.const_get(c.upcase))
+        expect(e).to               be_frozen
+        expect(e.enum_constant).to be_frozen
+      end
+    end
+
+    it "returns all values from the cache" do
+      expect(model.values.map(&:to_sym)).to contain_exactly(*CONSTANTS)
+    end
+  end
+
+  shared_examples "acts like a persisted enum" do
+    # abstract :model
+
+    include_examples "acts like an enum"
+
+    context "a referring model" do
+      let(:foreign_name) { model.model_name.singular }
+      let(:foreign_key) { foreign_name + "_id" }
+
+      let(:other_model) do
+        fn = foreign_name
+        create_test_model(:referrer, ->(t){ t.integer foreign_key }) do
+          belongs_to_enum fn
+        end
+      end
+
+      it "can be created from enum value" do
```

Comment:
These tests used to be close to tests that checked that the `belongs_to_enum` rejected invalid values. I can't find their equivalents.